packages/battery_plus/battery_plus/android/src/main/kotlin/dev/fluttercommunity/plus/battery/BatteryPlusPlugin.kt

```diff
@@ -156,12 +170,17 @@ class BatteryPlusPlugin : MethodCallHandler, EventChannel.StreamHandler, Flutter
         }
     }
 
-    private fun isXiaomiPowerSaveModeActive(): Boolean? {
-        val mode = Settings.System.getInt(applicationContext!!.contentResolver, POWER_SAVE_MODE_XIAOMI_NAME, -1)
+    @RequiresApi(VERSION_CODES.LOLLIPOP)
+    private fun isXiaomiPowerSaveModeActive(): Boolean {
+        val mode = Settings.System.getInt(
+            applicationContext!!.contentResolver,
+            POWER_SAVE_MODE_XIAOMI_NAME,
+            -1
+        )
         return if (mode != -1) {
             mode == POWER_SAVE_MODE_XIAOMI_VALUE
         } else {
-            null
+            checkPowerServiceSaveMode()
```

Comment:
On some newer android version of xiaomi devices, the power safe mode is not available by querying `POWER_SAVE_MODE_XIAOMI_NAME`.

This PR fallbacks to the default power safe mode. We are doing something similar for samsung as well.